packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -0,0 +1,494 @@
+      "Please use an empty unpartitioned Subscription with sessions for integration testing"
+    );
+  }
+}
+
+async function afterEachTest(): Promise<void> {
+  await ns.close();
+}
+describe("Accept a session without passing sessionId and receive messages - Queue/Subscription has messages belonging to same sessionId", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function testComplete_batching(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    const receiverClient = await sessionClient.acceptSession();
+    const msgs = await receiverClient.receiveBatch(1);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(msgs[0].body, testMessagesWithSessions[0].body);
+    should.equal(msgs[0].messageId, testMessagesWithSessions[0].messageId);
+    should.equal(msgs[0].deliveryCount, 0);
+    await msgs[0].complete();
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+
+  it("Partitioned Queues with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(partitionedQueueSessionClient, partitionedQueueSessionClient);
+  });
+
+  it("Partitioned Topics and Subscription with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(partitionedTopicSessionClient, partitionedSubscriptionSessionClient);
+  });
+
+  it("Unpartitioned Queues with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(unpartitionedQueueSessionClient, unpartitionedQueueSessionClient);
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+
+  async function testComplete_streaming(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    const receiverClient = await sessionClient.acceptSession();
+    const receivedMsgs: ServiceBusMessage[] = [];
+    receiverClient.receive((messageSession: MessageSession, msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      should.equal(msg.body, testMessagesWithSessions[0].body);
+      should.equal(msg.messageId, testMessagesWithSessions[0].messageId);
+      return Promise.resolve();
+    }, unExpectedErrorHandler);
+
+    for (let i = 0; i < 5; i++) {
+      await delay(1000);
+      if (receivedMsgs.length === testMessagesWithSessions.length) {
```

Comment:
`testMessagesWithSessions` is 2, but you are sending only 1 message. So this check will never be true

Reply:
Oh yes, I'll remove the loop and put some delay.